drgn/helpers/linux/mm.py

```diff
@@ -291,3 +294,26 @@ def environ(task: Object) -> List[bytes]:
     env_start = mm.env_start.value_()
     env_end = mm.env_end.value_()
     return access_remote_vm(mm, env_start, env_end - env_start).split(b"\0")[:-1]
+
+
+def find_slab(prog: Program, slab_name: str) -> Optional[Object]:
+    """
+    Return the slab object corresponding to slab_name
+
+    :param slab_name: name of the slab
+    :return: ``struct kmem_cache``
+    """
+    slab_caches_addr = prog["slab_caches"].address_of_().read_()
+    for s in list_for_each_entry("struct kmem_cache", slab_caches_addr, "list"):
```

Comment:
`address_of_()` already returns a value object, so you don't need `read_()` in this case. Also, since this iteration is duplicated in a couple of places, it would be nice to split it into its own helper. Something like

```python
def for_each_slab_cache(prog: Program) -> Iterator[Object]:
    return list_for_each_entry(
        "struct kmem_cache", prog["slab_caches"].address_of_(), "list"
    )
```

(Along with a matching test case, probably just comparing the set of names of the caches it returns against the set of names from `/proc/slabinfo`.)

Then you can use that for both `find_slab()` and `print_slabs()`.